bin/llvm-kompilex

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+set -e
+
+usage() {
+  cat << HERE
+Usage: $0 <definition.kore> <dt_dir> <object type> [options] -- [clang flags]
+   or: $0 <interpreter.o>            <object type> [options] -- [clang flags]
+
+Options:
+  -h, --help      Print this message and exit.
+  -v, --verbose   Print commands executed to stderr.
+  --save-temps    Do not delete temporary files on exit.
+
+Valid values for a KLLVM object type are:
+  main:     a main function will be generated. The resulting executable has the
+            signature "interpreter <input.kore> <depth> <output.kore>"
+
+  search:   as "main", but the resulting executable has collects all possible
+            results at each rewrite step.
+
+  static:   as "main", but do not link the resulting binary against the
+            necessary shared libraries. This must be done manually later.
+HERE
+}
+
+run () {
+  if [ "$verbose" = true ]; then
+    set -x
+  fi
+  "$@"
+  { set +x; } 2>/dev/null
+}
+
+positional_args=()
+reading_clang_args=false
+kompile_clang_flags=()
+clang_args=()
+
+verbose=false
+save_temps=false
+
+while [[ $# -gt 0 ]]; do
+  if [ "$reading_clang_args" = true ]; then
+    clang_args+=("$1")
+    shift
+    continue
+  fi
+
+  case $1 in
+    -h|--help)
+      usage
+      exit 0
+      ;;
+    -v|--verbose)
+      verbose=true
+      kompile_clang_flags+=("-v")
+      shift
+      ;;
+    --save-temps)
+      save_temps=true	    
+      kompile_clang_flags+=("-save-temps")
+      shift
+      ;;
+    --)
```

Comment:
I feel like in order to follow standard practices for scripts like this, this should probably be optional.